File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int curr=nums[0];
        for(int i=1;i<nums.size();i++){
            if (nums[i]==curr){
                return true;
            }
            curr=nums[i];
        }
        return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix=[1]
        suffix=[1]
        for num in nums[:-1]:
            prefix.append(num*prefix[-1])
        
        for num in reversed(nums[1:]):
            suffix.append(num*suffix[-1])

        suffix.reverse()

        sol=[]

        for (s,p) in zip(suffix,prefix):
            sol.append(s*p)

        return sol
        